### PR TITLE
Add authorizations for iscn and likenft

### PR DESCRIPTION
**Pull request by nnkken**
- [x] granting update permission on iscn id, without transfer ownership
- [x] granting modify permission on nft royalty config
- [x] granting update listing permission on nft marketplace
- [x] granting update offer permission on nft marketplace
- [x] granting update permission on nft class id
- [x] granting mint permission on nft class id
- [x] test cases for iscn authorizations
- [x] test cases for likenft authorizations

### CODE REVIEW

File: x/likenft/types/authz.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+func (a UpdateOfferAuthorization) ValidateBasic() error {
+	return nil
+}
+
+func NewUpdateClassAuthorization(classId string) *UpdateClassAuthorization {
+	return &UpdateClassAuthorization{
+		ClassId: classId,
+	}
+}
```
Comment (williamchong) [inline x/likenft/types/authz.go:102]: Please also support NewClass via ISCN ID

**Review by williamchong — CHANGES_REQUESTED** [review]
Want to add support for more message after reviewing, sorry for the trouble

File: x/likenft/types/authz.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+func NewMintNFTAuthorization(classId string) *MintNFTAuthorization {
+	return &MintNFTAuthorization{
+		ClassId: classId,
+	}
+}
+
+func (a MintNFTAuthorization) MsgTypeURL() string {
+	return sdk.MsgTypeURL(&MsgMintNFT{})
```
[the remark this answers is not in the document]
Comment (nnkken) [reply x/likenft/types/authz.go:130]: Class owner.
In authz, the authz module only validates the `MsgExec` wrapper, then the wrapped message is forwarded to the corresponding message handler, like normal message execution.
The `MsgMintNFT` message specifies who receives the NFT (by the `creator` field), so it's nothing related to the authz caller.

File: x/likenft/types/authz.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+var _ authz.Authorization = &MintNFTAuthorization{}
+
+func NewUpdateRoyaltyConfigAuthorization(classId string) *UpdateRoyaltyConfigAuthorization {
+	return &UpdateRoyaltyConfigAuthorization{
+		ClassId: classId,
+	}
+}
+
+func (a UpdateRoyaltyConfigAuthorization) MsgTypeURL() string {
+	return sdk.MsgTypeURL(&MsgUpdateRoyaltyConfig{})
```
[the remark this answers is not in the document]
Comment (nnkken) [reply x/likenft/types/authz.go:22]: Yes it throws

File: x/likenft/types/authz.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+func (a UpdateRoyaltyConfigAuthorization) ValidateBasic() error {
+	return nil
+}
+
+func NewUpdateListingAuthorization(classId string, nftId string) *UpdateListingAuthorization {
+	return &UpdateListingAuthorization{
+		ClassId: classId,
+		NftId:   nftId,
+	}
```
[the remark this answers is not in the document]
Comment (nnkken) [reply x/likenft/types/authz.go:44]: OK.
Do you think `empty NftId = allow all NFT IDs` is a good design?

[the remark this answers is not in the document]
Comment (williamchong) [reply x/likenft/types/authz.go:44]: Yes it would work

File: x/likenft/types/authz.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (williamchong) [reply x/likenft/types/authz.go:130]: We would need authz for MsgSend NFT too to use as combo for this to be useful then

**Review by williamchong — APPROVED** [review]

**Pull request merged**